### PR TITLE
Adding in Support for Flexible Session Keys

### LINKED ISSUE #66: Allow Configuration for Session Key
Hi all!

I'm currently building an authentication library, which ships with pre-built registration and login forms. Because of this, my library uses this csurf module to protect the form inputs.

My library initializes it's own session management stuff (to maintain user state), using the node-client-sessions library from Mozilla (https://github.com/mozilla/node-client-sessions), which allows me to specify a custom request session key.

So for instance, I can use that library to initialize ONE session as `req.stormpathSession`, and another as `req.session`.

I like doing this because this way -- my library does its own session stuff under `req.stormpathSession`, while the user USING my library can create / do their own session stuff as `req.session`, which is sort of the convention.

Anyhow: back to my original point. What I'd like is for this ilbrary to support custom session keys. This way, I could tell the csurf middleware to write data to `req.stormpathSession` instead of the default `req.session`. This would make it possible for my sort of use case to work.

If this is cool with ya'll, and you'll merge it in, I'd be happy to submit a clean patch supporting this behavior along with full tests / etc.

Let me know if this sounds OK!

### PR DESCRIPTION
**Pull request by rdegges**
This commit should fix #66.

Basically, I added a new option `sessionKey`, which allows the developer to specify the name of the session key used.

I added in README docs, a working implementation, and tests which support this.

Happy to make changes if needed.

### CODE REVIEW

**Comment by rdegges** [comment]
Hmm, so apparently client-sessions won't work with node 0.8. Not sure how to get around this for testing, since express-sessions don't allow for arbitrary session request keys. Any suggestions? Other than this, I'm sure things will pass.

**Comment by dougwilson** [comment]
We do need the tests to work on Node.js 0.8, and I don't forsee us dropping Node.js 0.8 from this module anytime soon. My only suggestion is perhaps just add a custom mini session module as a fixture in this source code instead of using another library?

File: package.json [hunk 1]
```diff
@@ -16,6 +16,7 @@
   },
   "devDependencies": {
     "body-parser": "~1.12.0",
+    "client-sessions": "^0.7.0",
```
Comment (dougwilson) [inline package.json]: Don't use "carret" versions.

**Comment by rdegges** [comment]
Thanks @dougwilson I'll get this fixed today ^^

**Comment by rdegges** [comment]
Woot, fixed. Apparently switching from ^ -> ~ was all that was required -- client-sessions *does* actually support 0.8.x.

Looks like we won't need any mocking libraries after all! Whew! Anything else need fixing, or does this look good to you?

**Comment by dougwilson** [comment]
Hi @rdegges , sorry for the delay. In general it's good. I'm just thinking over an idea that would make it even more flexible (just taking a function that will determine where the key will come from), but thinking that's probably better left for the next major to try and consolidate all these different options, rather than changing this PR :)

**Comment by rdegges** [comment]
Hey @dougwilson -- no problem! If you think that's a better option, I'm happy to accommodate. I do agree that passing a function in would be better for a future release though (or potentially even restructuring the options completely -- I have some ideas for that as well).

But not sure if you want in increment breaking changes atm or not.

**Comment by dougwilson** [comment]
Right. For now, I'm just going to take this as-is so people can put the session object at any property they choose :)

**Comment by rdegges** [comment]
Sounds good! I'll resolve those merge conflicts from the doc updates so you can merge this in then. Appreciate you taking the time to look this over ^^

**Comment by dougwilson** [comment]
> I'll resolve those merge conflicts from the doc updates so you can merge this in then. 

Cool :) It's not necessary, but I certainly welcome it. I'll have it merged in either way today, haha

**Comment by rdegges** [comment]
All fixed! Also: happy to help. I maintain a ton of projects myself, so I know it can get super crazy with PRs and merge conflicts and it takes a ton of time to sort stuff out.

Thanks for being so responsive! Has been a pleasure contributing tot his project ^^

**Comment by dougwilson** [comment]
haha, thanks :) I don't use the GitHub merge butt anyway, FWIW :)

**Comment by rdegges** [comment]
Thanks for merging this! Any chance you want to cut a release with this included? ^^

**Comment by dougwilson** [comment]
I will :) I just like to wait just a bit to reflect on it in case I realize there is some obvious issue. If you want to test it to at least confirm on that side, that would help :) Otherwise it'll likely be releases by default tomorrow.

**Comment by dougwilson** [comment]
@rdegges just published this to npm as version 1.8.0

**Comment by rdegges** [comment]
Thanks! I'm traveling so am slow to respond atm, but I really appreciate it! <33

**Pull request closed without merging**